Gtpu tunnel

mtirpak opened the pull request:
GTP-U tunnel support

mtirpak committed:
enable Rx/Tx callback in DPDK

mtirpak committed:
GTP-U tunnel support

Packets can be GTP-U encapsulated and decapsulated by configuring
the tunnel on a port as follows:

            tunnel:
              type: gtpu
              src_address: 10.0.0.1
              dst_address: 10.0.0.2
              data_file: /etc/gtpu-tunnels.csv

The csv file shall include the list of endpoint-ip,src-teid,dst-teid
tripples.

mtirpak committed:
added script for generating pdp context

The script adds the PDP contexts to EDA over gRPC and generates
the csv file for gtpu-tunnel configuration.

patrickjahns commented on src/platform_cfg.cpp:
```diff
@@ -296,6 +297,10 @@ void operator >> (const YAML::Node& node, CMacYamlInfo & mac_info) {
     if (! utl_yaml_read_uint16(node, "vlan", mac_info.m_vlan, 0, 0xfff)) {
         mac_info.m_vlan = 0;
     }
+
+    if (node.FindValue("tunnel")) {
+        mac_info.m_tunnel = Tunnel::Parse(node["tunnel"]);
+    }
```
```
src/platform_cfg.cpp.1.o: In function `operator>>(YAML::Node const&, CMacYamlInfo&)':
/opt/trex/linux/build/../../src/platform_cfg.cpp:302: undefined reference to `Tunnel::Parse(YAML::Node const&)'
collect2: error: ld returned 1 exit status
```
The change leads to build errors

patrickjahns replied:
Note this only happens when building `linux` and not `linux_dpdk`

mtirpak replied:
The GTP-U encap/decap patch depends on DPDK, it utilizes a DPDK TX/RX callback on the interface. Hence, the PR cannot be compiled without DPDK.

mtirpak committed:
try to load the vfio_pci driver in any case

Disabled the check for the iommu kernel parameter because
the AWS EC2 instance do not have iommu support.

mpastyl committed:
Randomize UDP source port of gtpu packets to make target utilize multiple workers

mpastyl committed:
Add comment to clarify previous use of src port will be overwritten

mpastyl committed:
UDP src port is now based on tunnel steid

mpastyl committed:
Merge pull request #2 from EMnify/gtpu-tunnel-randomize-udp-src

Randomize UDP src port used in GTPU encapsulation

alexschwartz commented:
<!-- emnify/github-curator/elderly-pr -->
Dear emnify contributor,
this PR seems to be rather old. Can we either close this PR, or merge it?
         
By the way, if the repo is no longer in use, we can consider to archive this repo (and close all open PRs beforehand).
         
In 4 weeks from now, this PR will be closed automatically.
         
Cheers
Alex
